qiskit_machine_learning/algorithms/trainable_model.py

```diff
@@ -196,3 +198,38 @@ def _choose_initial_point(self) -> np.ndarray:
         elif self._initial_point is None:
             self._initial_point = np.random.rand(self._neural_network.num_weights)
         return self._initial_point
+
+    def get_objective(
+        self,
+        function: ObjectiveFunction,
+        callback: Optional[Callable[[np.ndarray, float], None]] = None,
+    ) -> Callable:
+        """Returns a function to evaluates the objective at given weights.
+
+        This is the objective function to be passed to the optimizer that is used for evaluation.
+
+        Args:
+            function: The objective function whose objective is to be evaluated.
+            callback: a callable that can access the intermediate data during the optimization.
+                Two parameter values are passed to the callback as follows during each evaluation
+                by the optimizer for its current set of parameters as it works towards the minimum.
+                These are: the weights for the objective function, and the computed objective value.
+
+
+        Returns:
+            objective of the hamiltonian of each parameter, and, optionally, the expectation
+            converter.
+
+        Raises:
+            RuntimeError: If the circuit is not parameterized (i.e. has 0 free parameters).
```

Comment:
Unclear where `RuntimeError` is raised. I guess it is copy from `VQE` as well?

Reply:
Pardon, will be watchful.

Reply:
Rectified in https://github.com/Qiskit/qiskit-machine-learning/pull/151/commits/da9be27a354e830db4597e3ada21f6a593254ffc